packages/oid4vci-demo-frontend/src/components/SSIPrimaryButton/index.tsx

```diff
@@ -7,29 +7,46 @@ export interface IButtonProps {
     className?: string,
     disabled?: boolean,
     caption: string,
+    color?: string
     onClick?: () => void
 }
 
 const SSIPrimaryButton: React.FC<IButtonProps> = (props: IButtonProps) => {
     const ecosystem = useEcosystem()
     const config = ecosystem.getComponentConfig<SSIPrimaryButtonConfig>('SSIPrimaryButton')
     const mainContainerStyle = config.styles.mainContainer;
+    const buttonStyle = config.styles.button
     const {caption, disabled = false, onClick, style} = props
-    // TODO text
-    return (
-        <button style={{
-            height: 42,
-            width: 300,
-            borderRadius: 6,
-            background: `${mainContainerStyle.backgroundColor}`,
+    const color = props.color ?? '#FBFBFB'
+
+    const buildStyle = (): React.CSSProperties => {
+        const mergedStyle: React.CSSProperties = {
+            ...buttonStyle,
             ...style,
             ...(disabled && {opacity: 0.4}),
+            borderRadius: 6,
+            borderWidth: 0,
+            background: `${props.style?.backgroundColor ?? mainContainerStyle.backgroundColor ?? 'transparent'}`,
+            border: `1px solid ${color}`,
             display: 'flex',
             alignItems: 'center',
             justifyContent: 'center',
             cursor: 'pointer',
-            borderWidth: 0
-        }}
+        }
+
+        if (!mergedStyle.width) {
+            mergedStyle.width = 300
+        }
+        if (!mergedStyle.height) {
+            mergedStyle.height = 42
+        }
+        return mergedStyle
+    }
+
+
+    // TODO text
```

Comment:
do we still need to address this?

Reply:
Will do in separate branch <text> to <p>